Capri.Web/src/components/myProposals.vue

```diff
@@ -0,0 +1,290 @@
+						<span class="itemText">{{ item.studyType }}</span>
+					</template>
+					<template v-slot:item.freeSlots="{ item }">
+						<span class="itemText">
+							{{ item.maxStudents - item.freeSlots }} /
+							{{ item.maxStudents }}
+						</span>
+					</template>
+
+					<template v-slot:body.append>
+						<td @click="showEmptypopUp">
+							<span class="addItem">
+								<v-icon class="marginBottomSix"
+									>mdi-plus-box</v-icon
+								>
+								Add a new thesis
+							</span>
+						</td>
+					</template>
+				</v-data-table>
+			</v-col>
+		</v-row>
+	</v-container>
+</template>
+<script lang="ts">
+import { Vue, Component, Prop } from 'vue-property-decorator';
+import popUp from './popUp.vue';
+
+@Component({
+    components: {
+        popUp,
+    },
+})
+export default class MyProporsals extends Vue {
+    public popUp = {
+        show: false,
+        maxWidth: 1000,
+        data: {
+            title: {
+                text: '',
+                label: 'Thesis title',
+                type: 'editableTextField',
+                columns: 12,
+            },
+            thesisType: {
+                text: '',
+                label: 'Thesis type',
+                items: ['Bachelor', 'Master'],
+                chosen: '',
+                type: 'selectableField',
+                columns: 6,
+            },
+            studyType: {
+                text: '',
+                label: 'Study type',
+                items: ['Full-time', 'Part-time'],
+                chosen: '',
+                type: 'selectableField',
+                columns: 6,
+            },
+            students: {
+                text: '',
+                label: 'Student',
+                type: 'studentField',
+                columns: 12,
+                students: ['', ''],
+                max: 4,
+            },
+            description: {
+                text: '',
+                label: 'Description',
+                type: 'editableTextAreaField',
+                columns: 12,
+            },
+        },
+    };
+    public studyTypes = ['Full-time', 'Part-time'];
+    public thesisTypes = ['Bachelor', 'Master'];
+    public search = '';
+    public headers = [
+        {
+            sortable: true,
+            text: 'Title',
+            value: 'title',
+            width: '55%',
+            align: 'center',
+        },
+        {
+            sortable: true,
+            text: 'Degree',
+            value: 'thesisType',
+            width: '15%',
+            align: 'center',
+        },
+        {
+            sortable: true,
+            text: 'Type',
+            value: 'studyType',
+            width: '15%',
+            algin: 'center',
+        },
+        {
+            sortable: true,
+            text: 'State',
+            value: 'freeSlots',
+            width: '15%',
+            algin: 'center',
+        },
+    ];
+    public items = [
+        {
+            title: 'CaPri System',
```

Comment:
Rename/change fields:

title-> TopicPolish
maxStudents -> MaxNumberOfStudents
description -> Description
thesisType -> Level (StudyLevel, this is enum, 0 - Bachelor, 1 - Master)
studyType -> Mode (StudyMode, this is enum, 0 - FullTime, 1 - PartTime)
promoter -> PromoterId (Guid)
students -> Students (ICollection<Guid>)

Add fields:
Id (Guid)
TopicEnglish (string)
OutputData (string)
Specialization (string)
StartingDate (DateTime)
Status (ProposalStatus, this is enum, 0 - Taken, 1 - PartiallyTaken, 2 - Free)
StudyProfile (StudyProfile, this is enum, 0 - GeneralAcademic)
CourseId (Guid)

This is for ProposalViewModel